### PR TITLE
fix: promotion of delete notification template change

### PR DESCRIPTION
**Pull request by davidsoderberg**

### CODE REVIEW

File: apps/api/src/app/change/usecases/get-changes/get-changes.usecase.ts [hunk 1]
```diff
@@ -95,11 +95,19 @@ export class GetChanges {
...
       _id: entityId,
     });
 
+    if (!item) {
+      const items = await this.notificationTemplateRepository.findDeleted({
+        _id: entityId,
+        _environmentId: environmentId,
+      });
+      item = items[0];
+    }
```
Comment (djabarovgeorge) [inline apps/api/src/app/change/usecases/get-changes/get-changes.usecase.ts:109]: Could we do here one query findWithDeleted?

**Review by djabarovgeorge — APPROVED** [review]
Looks great :)

File: apps/api/src/app/change/usecases/get-changes/get-changes.usecase.ts (hunk 1 above)
Comment (davidsoderberg) [reply apps/api/src/app/change/usecases/get-changes/get-changes.usecase.ts:109]: I am not able to find it used in the project...

Comment (davidsoderberg) [reply apps/api/src/app/change/usecases/get-changes/get-changes.usecase.ts:109]: ah it is a part of the mongoose-delete package :)

Comment (djabarovgeorge) [reply apps/api/src/app/change/usecases/get-changes/get-changes.usecase.ts:109]: Yea sorry i needed to refer it is from it :)

File: apps/api/src/app/change/usecases/get-changes/get-changes.usecase.ts [hunk 2]
```diff
@@ -95,11 +95,19 @@ export class GetChanges {
     entityId: string,
     environmentId: string
   ): Promise<IViewEntity | Record<string, unknown>> {
-    const item = await this.notificationTemplateRepository.findOne({
+    let item = await this.notificationTemplateRepository.findOne({
       _environmentId: environmentId,
       _id: entityId,
     });
 
+    if (!item) {
```
Comment (scopsy) [inline apps/api/src/app/change/usecases/get-changes/get-changes.usecase.ts:103]: Could we maybe add an API e2e for this one?

Comment (davidsoderberg) [reply apps/api/src/app/change/usecases/get-changes/get-changes.usecase.ts:103]: I worked on it when you wrote this ;)

Comment (davidsoderberg) [reply apps/api/src/app/change/usecases/get-changes/get-changes.usecase.ts:103]: I added an expect for changes as well now :)

**Review by scopsy — APPROVED** [review]
Looks great!

**Pull request merged**